astropy/io/fits/fitstime.py

```diff
@@ -245,18 +356,52 @@ def _convert_time_column(col, column_info):
     """
 
     # The code might fail while attempting to read FITS files not written by astropy.
+    try:
+        # ISO-8601 is the only string representation of time in FITS
+        if col.info.dtype.kind in ['S', 'U']:
+            # [+/-C]CCYY-MM-DD[Thh:mm:ss[.s...]] where the number of characters
+            # from index 20 to the end of string represents the precision
+            precision = max(int(col.info.dtype.str[2:]) - 20, 0)
+            return Time(col, format='fits', scale=column_info['scale'],
+                        precision=precision,
+                        location=column_info['location'])
+
+        if column_info['format'] == 'gps':
+            return Time(col, format='gps', location=column_info['location'])
+
+        # If reference value is 0 for JD or MJD, the column values can be
+        # directly converted to Time, as they are absolute (relative
+        # to a globally accepted zero point).
+        if (column_info['ref_time']['val'] == 0 and
+            column_info['ref_time']['format'] in ['jd', 'mjd']):
+            # (jd1, jd2) where jd = jd1 + jd2
+            if col.shape[-1] == 2 and col.ndim > 1:
+                return Time(col[..., 0], col[..., 1], scale=column_info['scale'],
+                            format=column_info['ref_time']['format'],
+                            location=column_info['location'])
+            else:
+                return Time(col, scale=column_info['scale'],
+                            format=column_info['ref_time']['format'],
+                            location=column_info['location'])
+
+        # Reference time
+        ref_time = Time(column_info['ref_time']['val'], scale=column_info['scale'],
+                        format=column_info['ref_time']['format'],
+                        location=column_info['location'])
+
+        # Elapsed time since reference time
+        if col.shape[-1] == 2 and col.ndim > 1:
+            delta_time = TimeDelta(col[..., 0], col[..., 1])
+        else:
+            delta_time = TimeDelta(col)
 
-    # Read in time coordinate column as Time
-    if col.shape[-1] == 2 and col.ndim > 1:
-        col = Time(col[..., 0], col[..., 1], format='jd',
-                   scale=column_info['scale'],
-                   location=column_info['location'])
-    else:
+        return ref_time + delta_time
+    except Exception as err:
         warnings.warn(
-            'Time column {} is not in the astropy required (jd1, jd2) format. '
-            'Hence, it will not be read as astropy Time'.format(col.info.name),
+            'The exception "{}" was encountered while converting the time '
```

Comment:
Since we actually return the old column here, write the warning as "was encountered while trying to convert ..."